server/lib/nl/detection/place.py

```diff
@@ -295,13 +313,16 @@ def _get_stripped_string(query_parts: List[QueryPart],
 # recognized.
 #
 def _get_resolved_entities(query_parts: List[QueryPart]) -> List[Entity]:
-  entity_dcids = set()
+  seen_dcids = set()
+  entity_dcids = []
   for e in query_parts:
     if not e.dcids:
       continue
-    entity_dcids.update(e.dcids)
-
-  return _get_non_place_entities(list(entity_dcids), [])
+    new_dcids = [d for d in e.dcids if not d in seen_dcids]
+    seen_dcids.update(new_dcids)
+    # sort the dcids so that the list is deterministic
+    entity_dcids.extend(sorted(new_dcids))
```

Comment:
Should we keep the order in which user specified the list?  I "think" place entities maintains order

Reply:
So for example, if user asks for X, Y and Z, we know which one is first.

Reply:
yes, this still maintains order because it is sorting the dcids within each query part. In biomed, there will be lots of entities with the same name, so each query part can return multiple dcids & there's preference for showing the answer for all the dcids

Reply:
Hmm, then should we sort better maybe (a more common type ahead of less common one?).  Have a TODO if you think so too.

Reply:
I do agree with sorting better, but types probably won't do much because a lot of times, the entities have the same type (e.g., FGFR1 is the name for hg19_FGFR1 and hg38_FGFR1 which are both genes)>

But added a TODO to sort better